pkg/controller/control.go

```diff
@@ -1250,19 +1254,27 @@ func (c *Controller) handleDiskNoSpaceErrorForReplicas(replicaNoSpaceErrMap map[
 	}
 
 	replicaWritableCount := 0
+	writableReplicaNoSpaceList := []string{}
 	for _, r := range c.replicas {
-		if r.Mode == types.ERR {
+		if r.Mode != types.RW {
 			continue
 		}
+		if _, exist := replicaNoSpaceErrMap[r.Address]; exist {
+			writableReplicaNoSpaceList = append(writableReplicaNoSpaceList, r.Address)
+		}
 		replicaWritableCount++
 	}
 
 	// https://github.com/longhorn/longhorn/issues/10718#issuecomment-3190666485
 	// If a replica is on a disk that is out of space, and others are not, this replica will be marked as ERR to avoid the inconsistent data.
-	// The volume will be readable when all disks of the replicas are out of space so last one replica will not be marked as ERR because of ENOSPC.
-	if len(replicaNoSpaceErrMap) == replicaWritableCount {
+	// If all replicas are on disks that are out of space at the same time, the volume will be directly return an ENOSPC error and no replica will be set to ERR.
+	// Last one replica will not be marked as ERR for an ENOSPC error so the volume will still be readable when all disks of the replicas are out of space.
+	if len(replicaNoSpaceErrMap) == replicaWritableCount || (replicaWritableCount == 1 && len(writableReplicaNoSpaceList) == 1) {
```

Comment:
This logic does not make sense for the below case:

1. There are 1 RW replicas and 1 WO replica.
2. Only the WO replica encounters NoSpace error

In theory, Longhorn should mark the NoSpace replica as ERR and continue IO, rather than returning error `types.ErrNoSpaceLeftOnDevice` here